instance/management/commands/instance_statistics_csv.py

```diff
@@ -0,0 +1,294 @@
+        )
+
+    def get_appserver_statistics(self, playbook_output_dir, name_prefix, public_ip):
+        """ Execute the collect_activity playbook to gather statistics """
+        inventory = '[apps]\n{server}'.format(server=public_ip)
+        playbook_path = os.path.join(
+            settings.SITE_ROOT,
+            'playbooks/collect_activity/collect_activity.yml'
+        )
+
+        def log_line(line):
+            """Helper to pass to capture_playbook_output()."""
+            self.stderr.write(self.style.SUCCESS(line))
+        log_line.info = log_line
+        log_line.error = log_line
+
+        # Launch the collect_activity playbook, which places a file into the `playbook_output_dir`
+        # on this host.
+        ansible.capture_playbook_output(
+            requirements_path=os.path.join(os.path.dirname(playbook_path), 'requirements.txt'),
+            inventory_str=inventory,
+            vars_str=(
+                'local_output_dir: {output_dir}\n'
+                'local_output_filename: user_statistics\n'
+                'remote_output_filename: /tmp/activity_report\n'
+                'server_name_prefix: {server_name_prefix}\n'
+                'extra_script_arguments: --skip-hit-statistics'
+            ).format(
+                output_dir=playbook_output_dir,
+                server_name_prefix=name_prefix
+            ),
+            playbook_path=playbook_path,
+            username=settings.INSTANCE_LOGS_SERVER_SSH_USERNAME,
+            logger_=log_line,
+        )
+
+    def collect_instance_statistics(
+            self,
+            out,
+            domain_name,
+            start_date,
+            end_date
+    ):  # pylint: disable=too-many-locals
+        """Generate the instance statistics CSV."""
+        instance = self.get_instance_from_domain_name(domain_name)
+        appserver = instance.get_active_appservers().first()
+        name_prefix = appserver.server_name_prefix
+
+        self.stderr.write(self.style.SUCCESS('Running playbook...'))
+
+        with ansible.create_temp_dir() as playbook_output_dir:
+            self.get_elasticsearch_hits_data_summary(playbook_output_dir, name_prefix, start_date, end_date)
+            self.get_appserver_statistics(playbook_output_dir, name_prefix, appserver.server.public_ip)
+
+            csv_writer = csv.writer(out, quoting=csv.QUOTE_NONNUMERIC)
+            csv_writer.writerow([
+                'Fully Qualified Domain Name',
+                'Server Name Prefix',
+                'Name',
+                'Contact Email',
+                'Unique Hits',
+                'Total Hits',
+                'Total Courses',
+                'Total Users',
+                'Age (Days)'
+            ])
+
+            filenames = [os.path.join(playbook_output_dir, f) for f in os.listdir(playbook_output_dir)]
+            data = ConfigParser()
+            data.read(filenames)
+
+            try:
+                section = data[name_prefix]
```

Comment:
@pcockwell, the `stats.py` script is still creating output with the IP address as the section name and hence the values for 'courses', 'users' are set to `N/A'. Can you check and fix this?

Reply:
I think this will be fixed with the fix to `public_ip` above. How are you testing this so that I can try to confirm/debug?

Reply:
@pcockwell, I tested this on production with your PR branch and on debugging, found that the CSV file generated on the appserver (in `/tmp`) had its public IP address as the section name.

Reply:
Marking this as resolved as this has been fixed now.